src/ecs/resources/input/input_handler.rs

```diff
@@ -81,6 +28,8 @@ impl<'a> Iterator for Buttons<'a> {
 /// ```
 #[derive(Default)]
 pub struct InputHandler {
+    /// Maps inputs to actions and axes.
+    pub bindings: Bindings,
     pressed_keys: SmallVec<[VirtualKeyCode; 16]>,
```

Comment:
There are complications in doing that, it was definitely the first approach I considered though.  While it is possible to use enums as indices into an array it raises a few problems for maintenance purposes

- No built in alert system for when new enums are added that need to be accounted for.
- No guarantee that enums `as usize` values will continue to be contiguous in the future.

With this approach we have the downside that code may occasionally need to iterate a few values, which I consider to be preferable.